src/frontend/src/binder/insert.rs

```diff
@@ -102,6 +102,8 @@ impl Binder {
         returning_items: Vec<SelectItem>,
     ) -> Result<BoundInsert> {
         let (schema_name, table_name) = Self::resolve_schema_qualified_name(&self.db_name, name)?;
+        // bind insert table
+        self.context.clause = Some(Clause::Insert);
         self.bind_table(schema_name.as_deref(), &table_name, None)?;
```

Comment:
Maybe `bind_table` shall not be called at all so that its column names are not added into the context ... This is introduced by #7094 for `returning` and not sure if anything related breaks / conflicts.

The current fix also looks sensible.

Reply:
Binding into context is fine as long as we know where this context is generated.